tests/unit/template_stargate_test.go

```diff
@@ -311,20 +313,22 @@ var _ = Describe("Verify Stargate template", func() {
 		})
 
 		It("changing container image and imagePullPolicy", func() {
-			alternateImage := "stargateio/stargate-3_11:v1.0.3"
+			repository := "stargateio/stargate-3_11"
+			tag := "v1.0.3"
 			alternatePullPolicy := "Always"
 			options := &helm.Options{
 				KubectlOptions: defaultKubeCtlOptions,
 				SetValues: map[string]string{
-					"stargate.enabled":         "true",
-					"stargate.image":           alternateImage,
-					"stargate.imagePullPolicy": alternatePullPolicy,
+					"stargate.enabled":          "true",
+					"stargate.image.repository": repository,
+					"stargate.image.tag":        tag,
+					"stargate.image.pullPolicy": alternatePullPolicy,
 				},
 			}
 
 			Expect(renderTemplate(options)).To(Succeed())
 			container := deployment.Spec.Template.Spec.Containers[0]
-			Expect(container.Image).To(Equal(alternateImage))
+			Expect(container.Image).To(Equal("docker.io/" + repository + ":" + tag))
```

Comment:
Still concerned about the `docker.io` check here when we are not explicitly setting it.